datasets/un_ga/un_ga.py

```diff
@@ -0,0 +1,150 @@
+        """BuilderConfig for United nations general assembly resolutions: A six-language parallel corpus.
+        The first language in `language_pair` should consist of two strings joined by
+        an underscore (e.g. "en-tr").
+        Args:
+          language_pair: pair of languages that will be used for translation.
+          **kwargs: keyword arguments forwarded to super.
+        """
+        name = "%s_to_%s" % (language_pair[0], language_pair[1])
+
+        description = ("Translation dataset from %s to %s or %s to %s.") % (
+            language_pair[0],
+            language_pair[1],
+            language_pair[1],
+            language_pair[0],
+        )
+        super(UnGaConfig, self).__init__(
+            name=name, description=description, version=datasets.Version(_VERSION, ""), **kwargs
+        )
+
+        # Validate language pair.
+        assert language_pair in _VALID_LANGUAGE_PAIRS, (
+            "Config language pair (%s, " "%s) not supported"
+        ) % language_pair
+
+        self.language_pair = language_pair
+
+
+class UnGa(datasets.GeneratorBasedBuilder):
+
+    BUILDER_CONFIGS = [
+        UnGaConfig(
+            language_pair=pair,
+        )
+        for pair in _VALID_LANGUAGE_PAIRS.keys()
+    ]
+
+    BUILDER_CONFIG_CLASS = UnGaConfig
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "translation": datasets.Translation(languages=tuple(self.config.language_pair)),
+                },
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        download_url = _VALID_LANGUAGE_PAIRS.get(tuple(self.config.language_pair))
+        path = dl_manager.download_and_extract(download_url)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"datapath": path},
+            )
+        ]
+
+    def _generate_examples(self, datapath):
+        lang1, lang2 = self.config.language_pair
+        lang1_file = _BASE_NAME.format(lang1, lang2, lang1)
+        lang2_file = _BASE_NAME.format(lang1, lang2, lang2)
+        lang1_path = os.path.join(datapath, lang1_file)
+        lang2_path = os.path.join(datapath, lang2_file)
+
+        with open(lang1_path, encoding="utf-8") as f1, open(lang2_path, encoding="utf-8") as f2:
+            for sentence_counter, (x, y) in enumerate(zip(f1, f2)):
+                x = x.strip()
+                y = y.strip()
+                result = (
+                    sentence_counter,
+                    {
+                        "id": str(sentence_counter),
+                        "translation": {lang1: x, lang2: y},
+                    },
+                )
+                sentence_counter += 1
```

Comment:
You can remove this line

sentence_counter is already incremented thanks to enumerate